framework/meta/src/cargo_toml_contents.rs

```diff
@@ -55,6 +81,40 @@ impl CargoTomlContents {
             .to_string()
     }
 
+    pub fn package_edition(&self) -> String {
+        self.toml_value
+            .get("package")
+            .expect("missing package in Cargo.toml")
+            .get("edition")
+            .expect("missing package name in Cargo.toml")
+            .as_str()
+            .expect("package name not a string value")
+            .to_string()
+    }
+
+    pub fn mvx_dependency_version(&self) -> String {
+        let toml_value = self.dependency(FRAMEWORK_NAME_BASE).unwrap().to_owned();
+
+        if let Value::Table(table) = toml_value {
+            if let Some(version) = table.get("version") {
+                return remove_quotes(version);
+            }
+        }
+        panic!("could not find multiversx-sc dependency version in cargo toml")
+    }
+
+    pub fn mvx_dependency_path(&self) -> Option<String> {
```

Comment:
Let's make this more general, add an argument crate_name: &str, there's no need to hardcode the framework name here.